source/shared/cpp/ObjectModel/MarkDownParser.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <iomanip>
+#include <list>
+#include "pch.h"
+#include "BaseCardElement.h"
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+class MarkDownParser
+{
+public:
+    MarkDownParser (const std::string &txt); 
+
+    enum DelimiterType
+    {
+        Init,
+        Alphanumeric,
+        Puntuation,
+        Escape,
+        WhiteSpace,
+        Underscore = 0x8,
+        Asterisk   = 0x10,
+    };
+
+    enum EmphasisType
+    {
+        EmphasisItalic = 1,
+        EmphasisBold = 2,
+    };
+
+    enum EmphasisState
+    {   
+        // state for the first encounter of emphasis delimiter 
+        EmphasisStart = 0x1,  
+        // state for a emphasis delimiter run 
+        EmphasisRun = 0x2,   
+        InsideEmphasis = EmphasisStart | EmphasisRun,
+        // state for when a emphasis delimiter run is ended 
+        EmphasisEnd = 0x4,       
+        // init state for emphasis detection 
+        EmphasisNone = 0x8,
+        OutsideEmphasis = EmphasisEnd | EmphasisNone,
+        EndOfWords = 0x10,
+        BeginOfWords = 0x20,
+    };
+
+    struct Emphasis
+    {
+        int m_emphCnts;
+        unsigned m_idx;
```

Comment:
>m_idx [](start = 17, length = 5)

Would this make sense if renamed to position? if it doesn't, let's use the whole word: index

Reply:
it stores index of token table, so it makes more sense to keep it index; i will change it to whole word.

---
In reply to: [154744159](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154744159) [](ancestors = 154744159)